presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -19,23 +19,30 @@
 import com.facebook.presto.sql.planner.Symbol;
```

Comment:
does https://github.com/prestodb/presto/pull/8435 become obsolete? Can you copy tests from that PR to make sure they are working.